paddle/phi/infermeta/spmd_rules/fused_rope.cc

```diff
@@ -68,13 +68,35 @@ void check_k_or_v(const DistMetaTensor& k_or_v,
                                    ndim,
                                    dims_mapping_size));
 
+  int64_t k_num_head = shape[kNumHeadsDimIndex];
+  int64_t q_num_head = q_shape[kNumHeadsDimIndex];
   PADDLE_ENFORCE_EQ(
-      shape,
-      q_shape,
+      q_num_head % k_num_head == 0,
+      true,
       phi::errors::InvalidArgument(
-          "The shape of q and k/v's are not matched, [%d]  vs [%d]",
-          str_join(q_shape),
-          str_join(shape)));
+          "The num_head of q must be divisible by k/v's, but [%d]  vs [%d]",
```

Comment:
... by k and v, but got q_num_head is %d, k_num_head is %d

Reply:
Done